src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -624,13 +626,47 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                 case BoundKind.AssignmentOperator:
                     var assignment = (BoundAssignmentOperator)expr;
                     return CheckSimpleAssignmentValueKind(node, assignment, valueKind, diagnostics);
+
+                default:
+                    Debug.Assert(expr is not BoundValuePlaceholderBase, $"Placeholder kind {expr.Kind} should be explicitly handled");
+                    break;
             }
 
             // At this point we should have covered all the possible cases for anything that is not a strict RValue.
             Error(diagnostics, GetStandardLvalueError(valueKind), node);
             return false;
         }
 
+#nullable enable
+        private static BoundExpression? UnwrapPlaceholdersIfNeeded(BoundExpression? e)
```

Comment:
>UnwrapPlaceholdersIfNeeded [](http://example.com/codeflow?start=40&length=26)

It looks like there is only one call site for this method, consider inlining it, at least as a local function.